### PR TITLE
added side panel to select which documents, also includes dropdown for chunk size

### PR DESCRIPTION
**Pull request by aoalice**
## Describe your changes
image

- context variables to select which PDFs
- start of chunking selection (choosing recursive will open down the scrollbar for chunk size)

## Checklist before requesting a review
- [x] I have performed a self-review of my code
- [ ] If it is a core feature, I have added thorough tests.
- [x] Added necessary documentation and linked in PR

### CODE REVIEW

File: components/Context/index.tsx [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+export const Context: React.FC<ContextProps> = ({ className, selected, uploads }) => {
+  /*const [entries, setEntries] = useState([]);*/
+  console.log("UPLOADS ", uploads);
+  const [cards, setCards] = useState<ICard[]>([]);
+
+  const [splittingMethod, setSplittingMethod] = useState("markdown");
+  const [chunkSize, setChunkSize] = useState(256);
+  const [overlap, setOverlap] = useState(1);
+
+  // Scroll to selected card
```
Comment (snehasquasher) [inline components/Context/index.tsx:22]: delete this code since it is unused?

File: components/Context/index.tsx
Comment (snehasquasher) [inline components/Context/index.tsx:1]: can u rename this file to ChunkDropdown / naming that captures the functionality of the component

File: components/Context/urls.ts
Comment (snehasquasher) [inline components/Context/urls.ts:1]: delete file since unused

File: context/selected-context.ts
Comment (snehasquasher) [inline context/selected-context.ts:1]: change to camel casing: `selectedContext`

File: components/Context/UrlButton.tsx [hunk 2]
```diff
@@ -0,0 +1,51 @@
+// UrlButton.tsx
+
+import { Button } from "./Button";
+import React, { useContext, FC } from "react";
+import Link from "next/link";
+import selectedPDFsContext from "@/context/selected-context";
```
Comment (snehasquasher) [inline components/Context/UrlButton.tsx:6]: i dont think u need a separate component for selectedContext, u can just add that code into this file

File: components/Context/Card.tsx
Comment (snehasquasher) [inline components/Context/Card.tsx:1]: can u give the component a more descriptive name, ChunkSelectorCard for example

File: pages/chat.tsx [hunk 3]
```diff
@@ -27,8 +29,11 @@ export default function Chat() {
   const [faithfulnessScore, setFaithfulnessScore] = useState(0);
   const [contextRelevanceScore, setContextRelevanceScore] = useState(0);
   const [answerRelevanceScore, setAnswerRelevanceScore] = useState(0);
-  const [uploadedFileNames, setUploadedFileNames] = useState([]);
-  const [selectedFiles, setSelectedFiles] = useState([]);
+  const [uploadedFiles, setUploadedFiles] = useState([]);
+  const [context, setContext] = useState<string[] | null>(null);
```
Comment (snehasquasher) [inline pages/chat.tsx:33]: can u use a a more descriptive naming, unclear what exactly "setContext" refers to

File: pages/chat.tsx [hunk 4]
```diff
@@ -107,15 +113,17 @@ export default function Chat() {
...
       console.error("Selected files contain None values");
       return; // Don't proceed with the request
     }
+
+    console.log("selected PDFs for this response: ", selectedPDFs)
   
     let data = JSON.stringify({
       messages: [{ role: "user", content: oldInput }],
-      filenames: selectedFiles // Send array of selected filenames
+      filenames: selectedPDFs // Send array of selected filenames
```
Comment (snehasquasher) [inline pages/chat.tsx:126]: nice!

File: app.log
Comment (nijhawananushka) [inline app.log:1]: could you remove this file from the commit/add it to .gitignore?

**Review by nijhawananushka — APPROVED** [review]
LGTM

File: components/Context/index.tsx (hunk 1 above)
Comment (aoalice) [reply components/Context/index.tsx:22]: I think it'll be used for the scrollbar/chunk size selector? just hidden rn

**Pull request merged**